diag_manager/fms_diag_time_reduction.F90

```diff
@@ -0,0 +1,218 @@
+!***********************************************************************
+!*                   GNU Lesser General Public License
+!*
+!* This file is part of the GFDL Flexible Modeling System (FMS).
+!*
+!* FMS is free software: you can redistribute it and/or modify it under
+!* the terms of the GNU Lesser General Public License as published by
+!* the Free Software Foundation, either version 3 of the License, or (at
+!* your option) any later version.
+!*
+!* FMS is distributed in the hope that it will be useful, but WITHOUT
+!* ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+!* FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+!* for more details.
+!*
+!* You should have received a copy of the GNU Lesser General Public
+!* License along with FMS.  If not, see <http://www.gnu.org/licenses/>.
+!***********************************************************************
+
+!> @defgroup fms_diag_time_reduction_mod fms_diag_time_reduction_mod
+!> @ingroup diag_manager
+!> @brief fms_diag_time_reduction_mod defines classes encapsulating the diag_manager
+!! time redution types.
+!!
+!> @author Miguel Zuniga
+!!
+!> @file
+!> @brief File for @ref fms_diag_time_reduction_mod
+!> @addtogroup fms_diag_time_reduction_mod
+!> @{
+MODULE fms_diag_time_reduction_mod
+
+  USE diag_data_mod, only: EVERY_TIME
+   !!use diag_data_mod, only: time_min, time_max, time_sum, time_rms, time_average, time_none, time_power, &
+   !!& time_diurnal, every_time
+  USE fms_mod, ONLY: error_mesg, FATAL
+
+   implicit none
+
+   !!These parametes are the possible kinds of time reduction operations.
+   !!Note that sometimes one kind inplies another.
+   !!TODO: should they be put in diag_data ?
```

Comment:
That's where all the parameters are, but this is probably cleaner and easier to find.